TACC/Core-Styles
# fix: tup-488 scrollable table border

## Overview

Do not let the table border **above** all rows scroll away.

_The table border **below** all rows still scrolls into view. Solution unknown._

## Related

- [TUP-488](https://jira.tacc.utexas.edu/browse/TUP-488)
- replaces #263

## Changes

- **fixed** outdated comment
- **added** top-of-rows border via thead
- **changed** top-of-rows border via tbody to only be for headless table
- **added** comments to explain unique cell borders
- **changed** comment to be consistent

## Testing

1. Verify bug is fixed by testing [scrollable tables](http://localhost:3000/components/detail/o-fixed-header-table).
2. Verify other table styles are **not** changed:
    - http://localhost:3000/components/detail/table--default
    - http://localhost:3000/components/detail/table--cms
    - http://localhost:3000/components/detail/table--docs
    - _et cetera_

## UI

### 1. Bug Fixed

| Before \| After |
| - |
| ![Screenshot 2023-11-17 at 3 21 31 PM](https://github.com/TACC/Core-Styles/assets/62723358/32c7e260-7884-430d-a076-84a38df31388)

### 2. Styles Unchanged

<details>

| Before \| After |
| - |
| ![Screenshot 2023-11-17 at 3 17 40 PM](https://github.com/TACC/Core-Styles/assets/62723358/b29acc9b-da82-4329-9fc4-5207a4d9a6d8)
| ![Screenshot 2023-11-17 at 3 17 53 PM](https://github.com/TACC/Core-Styles/assets/62723358/0d55a345-0c2f-4aa9-8cc4-0f912d5d4ae3)
| ![Screenshot 2023-11-17 at 3 17 59 PM](https://github.com/TACC/Core-Styles/assets/62723358/8eea31b6-d91f-4eb0-8076-604ef586def1)
| ![Screenshot 2023-11-17 at 3 18 04 PM](https://github.com/TACC/Core-Styles/assets/62723358/f10d0ed0-35a6-4ffb-96f9-ba11fd9cbaee)
| ![Screenshot 2023-11-17 at 3 18 09 PM](https://github.com/TACC/Core-Styles/assets/62723358/a317aedd-e363-4673-afa1-77acdf3c3260)
| ![Screenshot 2023-11-17 at 3 18 14 PM](https://github.com/TACC/Core-Styles/assets/62723358/c5432f19-6736-4beb-9973-75474cb7b768)
| ![Screenshot 2023-11-17 at 3 18 19 PM](https://github.com/TACC/Core-Styles/assets/62723358/72efe2ab-34d3-4ab2-9777-d38d37cedb0c)
| ![Screenshot 2023-11-17 at 3 18 24 PM](https://github.com/TACC/Core-Styles/assets/62723358/114456af-7081-46fc-aa39-385c3b6935bc)
| ![Screenshot 2023-11-17 at 3 18 30 PM](https://github.com/TACC/Core-Styles/assets/62723358/7adfcb18-3a76-4c46-ad03-54cf288cb1a4)
| ![Screenshot 2023-11-17 at 3 18 36 PM](https://github.com/TACC/Core-Styles/assets/62723358/86fc339e-7aaa-4ede-a712-5846d4a0087d)
| ![Screenshot 2023-11-17 at 3 19 10 PM](https://github.com/TACC/Core-Styles/assets/62723358/0c003e6c-bb6b-458b-b317-ea41c862e858)
| ![Screenshot 2023-11-17 at 3 19 22 PM](https://github.com/TACC/Core-Styles/assets/62723358/044e3c12-5be1-4817-8238-166cec7c2d2f)
| ![Screenshot 2023-11-17 at 3 20 44 PM](https://github.com/TACC/Core-Styles/assets/62723358/ccffe1e2-7d60-4c87-ac86-4fd97ed6e59d)
| ![Screenshot 2023-11-17 at 3 20 55 PM](https://github.com/TACC/Core-Styles/assets/62723358/b13e7ce7-796f-43ec-83ad-995ead066e2f)
| ![Screenshot 2023-11-17 at 3 21 17 PM](https://github.com/TACC/Core-Styles/assets/62723358/bc22ae31-e392-47ba-b35f-908c6596330e)

</details>
